src/libertem/udf/base.py

```diff
@@ -1777,54 +1935,71 @@ def run_for_dataset_sync(
 
         executor = executor.ensure_sync()
 
-        damage = BufferWrapper(kind='nav', dtype=bool)
-        damage.set_shape_ds(dataset.shape, roi)
-        damage.allocate()
         try:
             if progress:
                 from tqdm import tqdm
                 t = tqdm(total=len(tasks))
             with executor.scatter(params) as params_handle:
                 if tasks:
-                    result_iter = executor.run_tasks(
+                    for res in executor.run_tasks(
                         tasks,
                         params_handle,
                         cancel_id,
-                    )
-                    for part_results, task in result_iter:
+                    ):
                         if progress:
                             t.update(1)
-                        for results, udf in zip(part_results, self._udfs):
-                            udf.set_views_for_partition(task.partition)
-                            udf.merge(
-                                dest=udf.results.get_proxy(),
-                                src=results.get_proxy()
-                            )
-                            udf.clear_views()
-                        v = damage.get_view_for_partition(task.partition)
-                        v[:] = True
-                        yield UDFResults(
-                            buffers=tuple(
-                                udf._do_get_results()
-                                for udf in self._udfs
-                            ),
-                            damage=damage
-                        )
-                else:
-                    # yield at least one result (which should be empty):
-                    for udf in self._udfs:
-                        udf.clear_views()
-                    yield UDFResults(
-                        buffers=tuple(
-                            udf._do_get_results()
-                            for udf in self._udfs
-                        ),
-                        damage=damage
-                    )
+                        yield res
         finally:
             if progress:
                 t.close()
 
+    def run_for_dataset_sync(
+        self,
+        dataset: DataSet,
+        executor: JobExecutor,
+        roi: Optional[np.ndarray] = None,
+        progress: bool = False,
+        corrections: Optional[CorrectionSet] = None,
+        backends: Optional[BackendSpec] = None,
+        dry: bool = False,
+        iterate: bool = True
```

Comment:
It really depends on what the user implements in `UDF.get_results()` and what partition size we have. It can be quite significant and is not in our control, that's why I'd skip it where possible.